pkg/storage/chunk/prometheus_chunk.go

```diff
@@ -106,15 +163,41 @@ func (p *prometheusChunkIterator) Value() model.SamplePair {
 	}
 }
 
-func (p *prometheusChunkIterator) Batch(size int) Batch {
+func (p *prometheusChunkIterator) Batch(size int, valueType chunkenc.ValueType) Batch {
 	var batch Batch
+	var populate func(j int)
+	switch valueType {
+	case chunkenc.ValFloat:
+		batch.ValueTypes = chunkenc.ValFloat
+		batch.SampleValues = &[BatchSize]float64{}
+		populate = func(j int) {
+			t, v := p.it.At()
+			batch.Timestamps[j] = t
+			batch.SampleValues[j] = v
+		}
+	case chunkenc.ValHistogram:
+		batch.ValueTypes = chunkenc.ValHistogram
+		batch.HistogramValues = &[BatchSize]*histogram.Histogram{}
+		populate = func(j int) {
+			t, h := p.it.AtHistogram()
+			batch.Timestamps[j] = t
+			batch.HistogramValues[j] = h
+		}
+	case chunkenc.ValFloatHistogram:
+		batch.ValueTypes = chunkenc.ValFloatHistogram
+		batch.FloatHistogramValues = &[BatchSize]*histogram.FloatHistogram{}
+		populate = func(j int) {
+			t, fh := p.it.AtFloatHistogram()
+			batch.Timestamps[j] = t
+			batch.FloatHistogramValues[j] = fh
+		}
+	}
```

Comment:
Should we have a default: panic case here?

Reply:
yes, adding